### PR TITLE
DEV-3916-bin-status

### PR DESCRIPTION
**Pull request by zamazzal**

### CODE REVIEW

File: sh/tests/bin-status_test.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+challenge() {
+	submitted=$(eval "$1" ; bash $FILENAME)
+	expected=$(eval "$1" ; bash solutions/bin-status.sh)
+
+	diff <(echo "$submitted") <(echo "$expected")
+}
+
+
+# True if FILE exists and is a regular file
+if [ -f ${FILENAME} ]; then
```
Comment (MSilva95) [inline sh/tests/bin-status_test.sh:20]: i have been testing a bit and found that if you use `echo "0"` as solution, the test will pass.

**Review by MSilva95 — CHANGES_REQUESTED** [review]

File: subjects/devops/bin-status/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+0
+$ false ; ./bin-status.sh
+1
+$
+```
+
+### Hints
+
+`$?` is a variable that holds the return value of the last executed command.
+`echo $?` displays 0 if the last command has been successfully executed and displays a non-zero value if some error has occurred.
```
[the remark this answers is not in the document]
Comment (zamazzal) [reply subjects/devops/bin-status/README.md:22]: yes, it's fine they are no challenge to print the last status, but the goal from this to teach the learner that they are a way to get the status code of last exec so he can use it to decide the next step

File: sh/tests/bin-status_test.sh (hunk 1 above)
Comment (zamazzal) [reply sh/tests/bin-status_test.sh:20]: in the test i test for different cases, return 0 and return != 0

File: subjects/devops/bin-status/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+## json-researcher
```
[the remark this answers is not in the document]
Comment (zamazzal) [reply subjects/devops/bin-status/README.md:1]: fixed

File: subjects/devops/bin-status/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+### Instructions
+
+Create the script `bin-status.sh` that will return the exit status of last command
+
+- What to use : `echo`
+
+- The output should be exactly like the example bellow but with the expected name
```
[the remark this answers is not in the document]
Comment (zamazzal) [reply subjects/devops/bin-status/README.md:9]: done

File: subjects/devops/bin-status/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+## json-researcher
+
+### Instructions
+
+Create the script `bin-status.sh` that will return the exit status of last command
```
[the remark this answers is not in the document]
Comment (zamazzal) [reply subjects/devops/bin-status/README.md:5]: done

File: subjects/devops/bin-status/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+- What to use : `echo`
```
[the remark this answers is not in the document]
Comment (zamazzal) [reply subjects/devops/bin-status/README.md:7]: done

File: subjects/devops/bin-status/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (MSilva95) [reply subjects/devops/bin-status/README.md:20]: understood

File: sh/tests/bin-status_test.sh (hunk 1 above)
Comment (MSilva95) [reply sh/tests/bin-status_test.sh:20]: Yes i see the test has some cases but it is returning  0
So if you put echo "0" the tests pass

**Review by MSilva95 — CHANGES_REQUESTED** [review]

File: sh/tests/bin-status_test.sh (hunk 1 above)
Comment (mikysett) [reply sh/tests/bin-status_test.sh:20]: fixed, see comment for the details

**Review by MSilva95 — APPROVED** [review]
LGTM

**Pull request merged**